### PR TITLE
Remove deprecated functionality for mlpack 3

### PR DESCRIPTION
**Pull request by rcurtin**
In my view this is the last thing that is necessary before branching off mlpack 3.  I also added any necessary licenses.

Is there anything else that needs to be added to HISTORY.md?  There has been so much development I am sure I missed some things.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -8,6 +8,21 @@
   * Add automatically generated Python bindings.  These have the same interface
     as the command-line programs.
 
+  * Add deep learning infrastructure in src/mlpack/methods/ann/.
```
Comment (zoq) [inline HISTORY.md]: and RL in rc/mlpack/methods/reinforcement_learning

File: HISTORY.md [hunk 2]
```diff
@@ -8,6 +8,21 @@
...
+
+  * Add hyperparameter tuning infrastructure and cross-validation infrastructure
+    in src/mlpack/core/cv/ and src/mlpack/core/hpt/.
+
+  * Fix bug in mean shift.
+
+  * Add random forests (see src/mlpack/methods/random_forest).
+
+  * Numerous other bugfixes and testing improvements.
+
```
Comment (zoq) [inline HISTORY.md]: We could add randomized and block krylov svd as well.

File: src/mlpack/methods/pca/pca.hpp [hunk 3]
```diff
@@ -40,8 +40,8 @@ class PCAType
    *
    * @param scaleData Whether or not to scale the data.
```
Comment (zoq) [inline src/mlpack/methods/pca/pca.hpp]: Should we remove the note from the file ``Note this class will be changed to have the name PCA in mlpack 3.0.0``?

**Comment by rcurtin** [comment]
Thanks for the comments, I've addressed them now.

**Pull request merged**